### DISCUSSION TITLE
Jetpack compose TextField value in State

### DISCUSSION THREAD

**Discussion by detsimov** [orbit-mvi/orbit-mvi#73] [Q&A]
Changing the value of a TextField to State through an intent is not possible. Since all intents run in a coroutine (probably) the TextField values ​​are out of sync for some reason, is there any solution to this problem?

**Sample code:** 

```
class MainActivity : ComponentActivity() {

    private val viewModel by viewModels<MainViewModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            ComposeSampleTheme {
                val state = viewModel.container.stateFlow.collectAsState()
                Surface(color = MaterialTheme.colors.background) {
                    TextField(
                        value = state.value.textFieldValue,
                        onValueChange = viewModel::valueChanged
                    )
                }
            }
        }
    }
}

class MainViewModel : ViewModel(), ContainerHost<MainViewModel.State, Nothing> {

    override val container: Container<State, Nothing> = viewModelScope.container(State())

    fun valueChanged(value: String) = intent {
        reduce { state.copy(textFieldValue = value) }
    }

    data class State(
        val textFieldValue: String = ""
    )
}
```

**Orbit version:** 4.2.0
**Compose version:** 1.0.1

**Comment by mattmook** [comment]
Hi @detsimov are you still seeing issues?

I'm currently running Compose 1.0.3 and I don't believe I'm seeing any issues.

I modified the example above to include a `Text(state.value.textFieldValue)` and am seeing it updating inline with the text changes.

https://user-images.githubusercontent.com/182422/137299288-df816d34-0533-45d0-9ece-6b340ee8594c.mp4
